ui/src/Components/SearchCandidates.jsx

```diff
@@ -37,63 +28,82 @@ export default class SearchCandidates extends React.Component {
 		.catch((error) => {console.log(error)});
 	}
 
-	/** Event Handlers for search fields
-		// TODO: Uncomment so state is updated, triggering a rerender of results
 	onNameSearch = (event) => {
-		this.setState({ searchName: event.target.value });
+		this.setState({ searchName: event.target.value.toLowerCase() });
+		console.log(event.target.value)
 		};
 
 	onEmailSearch = (event) => {
-		this.setState({ searchEmail: event.target.value });
+		this.setState({ searchEmail: event.target.value.toLowerCase() });
 		};
 
 	onNumberSearch = (event) => {
 		this.setState({ searchNumber: event.target.value });
 		};
 
-	*/
+	populate = (a, obj, i)=> {
+		a.push(<tr>
+			<th scope="row">{i+1}</th>
+			<td ><a href={"/profiles/"+obj[i].profile_id}>{obj[i].first_name} {obj[i].last_name}</a></td>
+			<td>{obj[i].email}</td>
+			<td>{obj[i].contact_num}</td>
+		</tr>)
+	}
 
 	render(){
-		console.log("RENDERING");
-		// Add results row by row
 		if (this.state.search) {
-			const obj = this.state.search.result; // instance of cleaner data queried earlier
+			const obj = this.state.search.result
+			console.log(obj[0])
 			var objLen = Object.keys(obj).length;
 			var items = [];
 
+
 			// Builds table with JSON object items passed in
 			for (var i = 0; i < objLen; i++) {
 				// !!! Link row to profile eventually
-				items.push(<tr>
-							<th scope="row">{i+1}</th>
-							<td ><a href={"/profiles/"+obj[i].profile_id}>{obj[i].first_name} {obj[i].last_name}</a></td>
-							<td>{obj[i].email}</td>
-							<td>{obj[i].contact_num}</td>
-						</tr>)
-			}
-
 
-
-			/** Filters json object with cleaner information, with filters specified in state
-				// TODO: Add logic to filter results when email & phone number values are typed in filters
-
-			var searchName = this.state.searchName;
-			var searchEmail = this.state.searchEmail;
-			var searchNumber = this.state.searchNumber;
-
-			function applyFilters() {
-				var finalResults = obj;
-				const fullName = String.prototype.toLowerCase.call(obj.first_name + ' ' + obj.last_name);
-				if (searchName != null) {
-					console.log(searchName);
-					finalResults = fullName.includes(String.prototype.toLowerCase.call(searchName));
+				//All of these if statements are necessary to make sure that the results returned are accurate. If we don't handle all of these cases, it will not populate dynamically
+				//These statements are also necessary to make sure that the filters work together, and not independently 
+				if (this.state.searchName != null && this.state.searchNumber != null && this.state.searchEmail != null) {
+					if ((obj[i].first_name.toLowerCase().includes(this.state.searchName) || obj[i].last_name.toLowerCase().includes(this.state.searchName)) && obj[i].contact_num.includes(this.state.searchNumber) && obj[i].email.toLowerCase().includes(this.state.searchEmail)) {
```

Comment:
I saw your comment about all the if statements being necessary, but I still feel like it's very redundant to have all this. I think we could have something like this:
```javascript
filters = []
if (this.state.searchName != null) {
    filters.push(searchName.toLowerCase());
}
// repeat for all filters...

// then make a call to a function to apply the filters
applyFilters(filters);

Reply:
the populate() function applies the filters. The only thing is I need to assess every case or else the filters act independently. An example of this is that first if statement, where I'm seeing if ALL fields are populated. Here, the filters need to work together, and what you type in for the name needs to be further filtered by what is typed in for the phone number.

If I consider these separately, they would be showing data that satisfies either or, not both together. I also need that final "else" statement so that it shows the overall data before any fields are inputted into the filters.

Reply:
^ your logic makes sense
I worry that once we start adding the additional filters in the future sprints, creating these matrices of cases might become impossible. Are there any other ideas you had for future refactoring?

Reply:
@christinajiyunlee I'll have to give it some thought! Possibly a method with optional parameters, but would have to dig a little deeper on my end since not too familiar with JS :)